omigami/data_structures/data_models.py

```diff
@@ -115,5 +115,10 @@ class ScoreCurve:
 
 @dataclass
 class FeatureSelectionResults:
+    raw_results: FeatureSelectionRawResults
     selected_features: SelectedFeatures
-    score_curve: ScoreCurve
+    score_curves: Dict[str, List[ScoreCurve]]
+    selected_feature_names: SelectedFeatures = None
```

Comment:
very minor: you could set it as a @property and enforce a functional relationship between selected_features and feature_names

Reply:
I think it is nice to have both separate here instead of one containing possibly indexes and names.. Do you see a problem in it?

Reply:
It was just a way to enforce consistency, but as I said very minor, don't warry